clang/lib/Index/IndexingAction.cpp

```diff
@@ -101,6 +103,8 @@ class IndexASTConsumer final : public ASTConsumer {
   std::shared_ptr<IndexingContext> IndexCtx;
   std::shared_ptr<Preprocessor> PP;
   std::function<bool(const Decl *)> ShouldSkipFunctionBody;
+  bool DeferIndexingToEndOfTranslationUnit;
+  std::vector<DeclGroupRef> TopLevelDecls;
```

Comment:
I don't think we need this. Instead, in `Initialize` we can save the `ASTContext*` to a field, and in `HandleTranslationUnit`, we can do:

```c++
for (auto *D : Context->getTranslationUnitDecl()->decls()) {
  IndexCtx->indexTopLevelDecl(D);
}
```

Reply:
Ah, right, of course. I overlooked that.